src/indexer/operation.rs

```diff
@@ -1,22 +1,38 @@
+use itertools::Either;
+
+use crate::query::Weight;
 use crate::schema::{Document, Term};
 use crate::Opstamp;
 
 /// Timestamped Delete operation.
-#[derive(Clone, Eq, PartialEq, Debug)]
+//#[derive(Clone, Debug)]
 pub struct DeleteOperation {
     pub opstamp: Opstamp,
-    pub term: Term,
+    pub term: Either<Term, Box<dyn Weight>>,
```

Comment:
Using `Either` here seems a bit "type blind" since it does not explain what the alternatives are. Why not create a new enum like

```rust
pub enum DeleteTarget {
  Term(Term),
  Query(Box<dyn Weight),
}
```

I think this would also avoid the need for the constructor methods as the literal syntax would stay readable, e.g.

```rust
DeleteOperation {
  opstamp: i as u64,
  target: DeleteTarget::Term(Term::from_field_u64(field, i as u64)),
}
```

Reply:
(Alternatively, I would suggest `DeleteOperation::from_term` or `DeleteOperation::from_query` as consistent names for the constructors.)